pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -235,6 +235,9 @@ func (r *ReconcileScanSettingBinding) Reconcile(ctx context.Context, request rec
 	found := compliancev1alpha1.ComplianceSuite{}
 	err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: suite.Namespace, Name: suite.Name}, &found)
 	if errors.IsNotFound(err) {
+		suiteTypeMeta := metav1.TypeMeta{}
+		suiteTypeMeta.SetGroupVersionKind(compliancev1alpha1.SchemeGroupVersion.WithKind("ComplianceSuite"))
```

Comment:
Because we have to specify this now, I'm wondering what happens to the rest of the occurrences across our controllers. This one just happen to be asserted in the tests, which is why I needed to add it here.

Reply:
I tested this on a cluster with the latest 1.5.0 release and this branch. I created an SSB, and then inspected the resulting suite to make sure they were the same (sans timestamps and generated IDs).

```diff
4c4
<   creationTimestamp: "2024-07-26T00:53:10Z"
---
>   creationTimestamp: "2024-07-26T01:02:52Z"
16,18c16,18
<     uid: 5b6ae8a4-06a9-44b9-82bc-489ecf47d28a
<   resourceVersion: "35727"
<   uid: 4c3f046c-1997-4ddb-963c-197e1f837a9a
---
>     uid: cda43572-16bd-4068-9ed9-b0eec22736ba
>   resourceVersion: "43510"
>   uid: f4a45365-e86d-44d7-9e80-cfbd1a4cd14d
22c22
<snip>
```

The rest were timestamps and image location changes. I think what we may have experienced here is just a gap in testing the Reconcile loop directly. From an API perspective, this change hasn't introduced a regression from what I can tell.

Reply:
@rhmdnd I wonder if we need to set it because we didn't when `suite` is defined on Line 134.
It doesn't have `TypeMeta: metav1.TypeMeta{Kind: "ComplianceSuite", APIVersion: ....}`
https://github.com/ComplianceAsCode/compliance-operator/pull/523/files#diff-c647a52d3b0966655cdd2384a18bfdaa5e8d694df700395aabba380140b89ef2R134

I see in other places that we define it when objects are created from scratch, for example:
- https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/profileparser/profileparser.go#L472
- https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/profileparser/profileparser.go#L678
- https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/controller/compliancescan/rawresults.go#L72
- https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/controller/tailoredprofile/tailoredprofile_controller.go#L688

Reply:
Nice catch - I added that approach so we're consistent with the references you linked.